.github/workflows/gradle.yml

```diff
@@ -101,14 +112,14 @@ jobs:
         uses: actions/cache@v2.1.0
         with:
           path: ${{env.BUILD_CACHE_PATH}}
-          key: ${{github.run_id}}
+          key: ${{github.ref}}
```

Comment:
What is the effect of this change? `run_id` would cache the output for re-runs of the same workflow.

Reply:
Now instead of being tied to a github actions run, it's tied to the source. (PR or Branch). This means that multiple pushes to the same branch can avoid re-executing if the tests for that target passed and no related code was changed just like the do locally.